cpp/oneapi/dal/algo/knn/backend/cpu/train_kernel_kd_tree_classification.cpp

```diff
@@ -68,7 +68,7 @@ static train_result<task::classification> call_daal_kernel(const context_cpu& ct
     interop::status_to_exception(status);
 
     auto knn_model = static_cast<daal_knn::Model*>(model_ptr.get());
-    const bool copy_data_labels = true;
+    const bool copy_data_labels = data_use_in_model == daal_knn::doNotUse;
```

Comment:
The data passed to DAAL kernels are the copy of users' data passed through new interfaces into this function. So we can use this data directly in the model since it is our internal copy. If we set `copy_data_labels = true` when we will have two data copies in this function (!): first to pass data to the kernels, and second to pass them to the model.
Which part of this logic is unclear to you?

Reply:
The problem is that we always copy user's table as homogeneous, however the copy should preserve table type, layout, information about features, etc. 

So the copy logic
```c++
auto arr_data = row_accessor<const Float>{ data }.pull();
const auto daal_data = interop::convert_to_daal_homogen_table(arr_data, row_count, column_count);
```

should be changes in the future.

I suggest to put TODO here to not forget it, I'll file this problem to the planning table.